core/toolbox/toolbox_events.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Class for a toolbox change event.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.Events.ToolboxChange');
+
+goog.require('Blockly.Events');
+goog.require('Blockly.Events.Abstract');
+goog.require('Blockly.registry');
+goog.require('Blockly.utils.object');
+
+
+/**
+ * Class for a toolbox changed event.
+ * Used to notify the developer when a user has clicked on a new toolbox item.
+ * @param {?string} oldType The type of the previously selected toolbox item.
+ * @param {?string} oldId The id of the previously selected toolbox item.
+ * @param {?string} newType The type of the currently selected toolbox item.
+ * @param {?string} newId The id of the currently selected toolbox item.
+ * @param {Blockly.Workspace} workspace The workspace the toolbox is on.
+ * @extends {Blockly.Events.Abstract}
+ * @constructor
+ */
+Blockly.Events.ToolboxChange = function(oldType, oldId, newType, newId, workspace) {
+  /**
```

Comment:
Will also have to add a way to determine that the block is indeed blank, checking that the first parameter is undefined should suffice: 
```
  this.isBlank = typeof opt_oldId == 'undefined';
```

Could also consider passing in the toolbox as the first parameter since this is a toolbox event.

Reply:
done and also passed in the toolbox.